hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java

```diff
@@ -69,15 +70,41 @@ public record PreHandleResult(
         @Nullable PreHandleResult innerResult,
         long configVersion) {
 
-    public Set<Key> getRequiredKeys() {
+    /**
+     * Returns whether this result's verification results are valid for the given context. This is <b>only</b>
+     * true if all keys linked to the transaction are exactly the same as those determined to be necessary
+     * in the given context.
+     *
+     * <p>Any change at all in a linked key means we must re-compute the verification results.
+     *
+     * @param context the context that might be able to reuse our verification results
+     * @return whether our verification results are reusable in the given context
+     */
+    public boolean hasReusableVerificationResultsFor(@NonNull final PreHandleContext context) {
+        return getPayerKey().equals(context.payerKey())
+                && getRequiredKeys().equals(context.requiredNonPayerKeys())
+                && getOptionalKeys().equals(context.optionalNonPayerKeys())
+                && getHollowAccounts().equals(context.requiredHollowAccounts());
+    }
+
+    /**
+     * Returns the key verifications for this result; or an empty map if none could be computed.
+     *
+     * @return the key verifications for this result; or an empty map if none could be computed.
```

Comment:
Duplicated comment lines.

Reply:
True, I think it's ok to omit the `@return` javadoc here; but repeating it is also acceptable I believe. 🤔